cli/tools/bundle/mod.rs

```diff
@@ -0,0 +1,879 @@
+    workspace_factory.workspace_npm_patch_packages()?,
+    installer_factory.tarball_cache()?,
+    factory.npm_cache()?,
+    "darwin-arm64",
+  )
+  .await?;
+  let resolver = factory.resolver().await?.clone();
+  let module_load_preparer = factory.module_load_preparer().await?.clone();
+  let root_permissions = factory.root_permissions_container()?;
+  let npm_resolver = factory.npm_resolver().await?.clone();
+  let node_resolver = factory.node_resolver().await?.clone();
+  let cli_options = factory.cli_options()?;
+  let module_loader = factory
+    .create_module_loader_factory()
+    .await?
+    .create_for_main(root_permissions.clone())
+    .module_loader;
+  let sys = factory.sys();
+  let init_cwd = cli_options.initial_cwd().canonicalize()?;
+
+  let plugin_handler = Arc::new(DenoPluginHandler {
+    resolver: resolver.clone(),
+    module_load_preparer,
+    module_graph_container: factory
+      .main_module_graph_container()
+      .await?
+      .clone(),
+    permissions: root_permissions.clone(),
+    npm_req_resolver: factory.npm_req_resolver()?.clone(),
+    npm_resolver: npm_resolver.clone(),
+    node_resolver: node_resolver.clone(),
+    http_cache: factory.http_cache()?.clone(),
+    module_loader: module_loader.clone(),
+  });
+  let start = std::time::Instant::now();
+
+  let entrypoint = bundle_flags
+    .entrypoints
+    .first()
+    .iter()
+    .map(|e| resolve_url_or_path(e, &init_cwd).unwrap())
+    .collect::<Vec<_>>();
+  let resolved = {
+    let mut resolved = vec![];
+    let init_cwd_url = Url::from_directory_path(&init_cwd).unwrap();
+    for e in &entrypoint {
+      let r = resolver
+        .resolve(
+          e.as_str(),
+          &init_cwd_url,
+          Position::new(0, 0),
+          ResolutionMode::Import,
+          NodeResolutionKind::Execution,
+        )
+        .unwrap();
+      resolved.push(r);
+    }
+    resolved
+  };
+
+  let roots = resolved
+    .into_iter()
+    .map(|url| {
+      if let Ok(v) = NpmPackageReqReference::from_specifier(&url) {
+        let referrer =
+          ModuleSpecifier::from_directory_path(sys.env_current_dir().unwrap())
+            .unwrap()
+            .join("package.json")
+            .unwrap();
+        let package_folder = npm_resolver
+          .resolve_pkg_folder_from_deno_module_req(v.req(), &referrer)
+          .unwrap();
+        let main_module = node_resolver
+          .resolve_binary_export(&package_folder, v.sub_path())
+          .unwrap();
+        Url::from_file_path(&main_module).unwrap()
+      } else {
+        url
+      }
+    })
+    .collect::<Vec<_>>();
+  let _ = plugin_handler.prepare_module_load(&roots).await;
+  let esbuild =
+    EsbuildService::new(esbuild_path, ESBUILD_VERSION, plugin_handler.clone())
+      .await
+      .unwrap();
+  let client = esbuild.client().clone();
+
+  {
+    tokio::spawn(async move {
+      loop {
+        tokio::select! {
+            res = esbuild.wait_for_exit() => {
```

Comment:
This seems fishy and I'm surprised `tokio::select!` doesn't complain there's there's only a single arm. It can just be a plain `loop` right?